### DISCUSSION TITLE
Is supabase auth production ready?

### DISCUSSION THREAD

**Discussion by ydennisy** [supabase/supabase#5627] [Questions] [answered]
We have been switching from firebase to supabase auth for a week and have been running into many small issues which have required some hacky workarounds.

Our users and accounts system in not the simplest (not the most complex) we have accounts and account types, and also are using the invite and google + password auth.

We have needed to apply our own values in localStorage to keep track of various states, something which we did not need to do previously.

So I am trying to understand if this is our poor implementation of if there are still some issues with supabase auth for non-trivial use cases.

**Comment by visudo20179C** [comment]
I had a similar issue where I wasn't able to toggle an auth state in my pinia store because after redirect from the OAuth provider, the value would just get overwritten. To workaround this I just put the `onAuthStateChange` hook in my App.vue component like so.

```
supabase.auth.onAuthStateChange(async (event) => {
    if (event !== 'SIGNED_OUT') {
        // handle sign in
    } else {
        // handle sign out
    }
});
```

This seems to allow the values in our pinia store to persist even after the redirect from the OAuth provider. 

With this you can at least keep track of users' login state and their related data with the `supabase.auth.user()` method.

Let me know if this helps at all. I'm using Vue3 and supabase.js client.

**Comment by berik** [comment]
Hello @thorwebdev I would like to know if you recommend to use supabase auth helper in production or is it still under active development? 

PS: I’ve watched this [YouTube video](https://youtu.be/0Fs96oZ4se0) and found your username

**Reply by hf** [reply]
Yes, you can use Auth Helpers in production. The beta status means that there could be breaking changes, especially when looking at Next.js 13 support, the approach will likely be somewhat different and will require a migration. But we're using semantic versioning so breaking changes will be behind a major version change and we will provide a migration guide.

**Comment by hf** [comment, marked as answer]
> Is supabase auth production ready?

Yes. We dogfood it on app.supabase.com as well and it's being actively developed and maintained.

> So I am trying to understand if this is our poor implementation of if there are still some issues with supabase auth for non-trivial use cases.

Depends what your use cases are. I'd love to get feedback on what your use case looked like. I'd appreciate it if you could go into detail about your biggest challenges.

**Comment by l16i** [comment]
@hf  thank you for your feedback.
I'm totally new to this project and I would really like to use it in production app. Just to be sure when you said

> you can use Auth Helpers in production. The beta status means that there could be breaking changes

Did you also imply that all the feature of Supabase can be used for production or only the Auth (and some other specific one)?

Except the breaking changes in the API should I be worried about something else?
Since the readme still doesn't check the "production ready" item

Thanks
